File: app/routes/admin_api/equipment.py
from app.models.equipment import Equipment
from app.schemas.equipment import EquipmentSchema
from app.decorators import admin_required
from app.routes.admin_api import admin_api_bp
from app.routes.admin_api.crud_helpers import (
    paginated_list, get_or_404_json, create_item, update_item,
    delete_item, bulk_delete,
)

equipment_schema = EquipmentSchema()
equipment_list_schema = EquipmentSchema(many=True)

UPDATABLE_FIELDS = ['name', 'description', 'image_url', 'category', 'specs', 'is_available', 'order']
SEARCH_FIELDS = ['name', 'description', 'category']
FILTER_FIELDS = [
    ('is_available', 'is_available', True),
    ('category', 'category', False),
]


@admin_api_bp.route('/equipment', methods=['GET'])
@admin_required
def list_equipment():
    return paginated_list(Equipment, equipment_list_schema, search_fields=SEARCH_FIELDS,
                          filter_fields=FILTER_FIELDS, default_sort='order', default_order='asc')


@admin_api_bp.route('/equipment/<int:equipment_id>', methods=['GET'])
@admin_required
def get_equipment(equipment_id):
    return get_or_404_json(Equipment, equipment_id, equipment_schema)


@admin_api_bp.route('/equipment', methods=['POST'])
@admin_required
def create_equipment():
    return create_item(Equipment, equipment_schema, 'equipment')


@admin_api_bp.route('/equipment/<int:equipment_id>', methods=['PUT'])
@admin_required
def update_equipment(equipment_id):
    return update_item(Equipment, equipment_schema, equipment_id, 'equipment', UPDATABLE_FIELDS)


@admin_api_bp.route('/equipment/<int:equipment_id>', methods=['DELETE'])
@admin_required
def delete_equipment(equipment_id):
    return delete_item(Equipment, equipment_id, 'equipment', 'Оборудование удалено')


@admin_api_bp.route('/equipment/bulk-delete', methods=['POST'])
@admin_required
def bulk_delete_equipment():
    return bulk_delete(Equipment, 'equipment')

File: admin-panel/src/api/equipment.ts
import client from './client'
import type { PaginatedResponse, ListParams, SingleResponse } from '@/types/api'

export interface Equipment {
  id: number
  name: string
  description: string | null
  image_url: string | null
  category: string | null
  specs: string | null
  is_available: boolean
  order: number
}

export const equipmentApi = {
  list: (params?: ListParams) =>
    client.get<PaginatedResponse<Equipment>>('/equipment', { params }).then((r) => r.data),

  get: (id: number) =>
    client.get<SingleResponse<Equipment>>(`/equipment/${id}`).then((r) => r.data.data),

  create: (data: Partial<Equipment>) =>
    client.post<SingleResponse<Equipment>>('/equipment', data).then((r) => r.data.data),

  update: (id: number, data: Partial<Equipment>) =>
    client.put<SingleResponse<Equipment>>(`/equipment/${id}`, data).then((r) => r.data.data),

  delete: (id: number) => client.delete(`/equipment/${id}`),

  bulkDelete: (ids: number[]) => client.post('/equipment/bulk-delete', { ids }),
}
